pkg/controller/utils/utils.go

```diff
@@ -113,3 +116,21 @@ func Remove(list []string, s string) []string {
 	}
 	return list
 }
+
+func ValidateAWSclient(reqLogger logr.Logger, awsClient awsclient.Client) error {
+	for retry := 1; retry <= 6; retry++ {
+		_, err := awsClient.GetUser(&iam.GetUserInput{})
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				if aerr.Code() == "InvalidClientTokenId" {
+					reqLogger.Info(fmt.Sprintf("New credentials not yet available after rotating. Attempt %b/6", retry))
+				} else {
+					reqLogger.Error(err, "Failed to create client with rotated credentials")
+					return err
+				}
+			}
+			time.Sleep(10 * time.Second)
```

Comment:
10 seconds is a long time to wait, the reconcile loop will stop and wait here. Could we change this to something like;

```
duration := 500 * time.Duration(retry) * time.Millisecond
time.Sleep(duration)
```

Reply:
Yes. good spot, will update this.